plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcConnectionCreation.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import io.trino.plugin.jdbc.credential.CredentialProvider;
+import io.trino.plugin.jdbc.credential.EmptyCredentialProvider;
+import io.trino.plugin.jdbc.mapping.IdentifierMapping;
+import io.trino.testing.QueryRunner;
+import org.h2.Driver;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+import java.util.Properties;
+
+import static io.trino.plugin.jdbc.H2QueryRunner.createH2QueryRunner;
+import static io.trino.plugin.jdbc.TestingH2JdbcModule.createH2ConnectionUrl;
+import static io.trino.tpch.TpchTable.NATION;
+import static io.trino.tpch.TpchTable.REGION;
+import static java.util.Objects.requireNonNull;
+
+@Test(singleThreaded = true) // inherited from BaseJdbcConnectionCreationTest
+public class TestJdbcConnectionCreation
+        extends BaseJdbcConnectionCreationTest
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        String connectionUrl = createH2ConnectionUrl();
+        DriverConnectionFactory delegate = new DriverConnectionFactory(new Driver(), connectionUrl, new Properties(), new EmptyCredentialProvider());
+        this.connectionFactory = new ConnectionCountingConnectionFactory(delegate);
+        return createH2QueryRunner(ImmutableList.of(NATION, REGION), ImmutableMap.of("connection-url", connectionUrl), new TestingConnectionH2Module(connectionFactory));
+    }
+
+    @Test(dataProvider = "testCases")
+    public void test(@Language("SQL") String query, int expectedJdbcConnectionsCount, Optional<String> errorMessage)
+    {
+        assertJdbcConnections(query, expectedJdbcConnectionsCount, errorMessage);
+    }
+
+    @DataProvider
+    public Object[][] testCases()
+    {
+        return new Object[][] {
+                {"SELECT * FROM nation LIMIT 1", 3, Optional.empty()},
```

Comment:
Just thoughs aloud: these are trino statements, and reused for different connectors, however even for h2 and postgres numbers are really different. Would be great to reuse them somehow..

Reply:
Queries are the almost the same, however expected behavior is connector specific. I don't see a good way of doing this and keeping the code simple.